### DISCUSSION TITLE
Is there a way to remove all images from a PDF file.

### DISCUSSION THREAD

**Discussion by megh-khaire** [pymupdf/PyMuPDF#2283] [Looking for help] [answered]
Hello, I'm searching for a way of removing all images from a PDF file along with the labels printed on these images.

**Comment by JorjMcKie** [comment, marked as answer]
Hm, removing images is possible in more than one way.
What does "labels printed on these images" mean?

The easiest way to remove images probably is using redaction annotations: loop over the pages, detect all image boundary boxes on it, add a redaction annot for each of the bboxes and then apply the redactions.
... not forgetting to use a suitable garbage collection option on save.

If the "labels" refers to captions of text shown underneath an image: this is not encoded as such in PDF. You must find your own way to locate this text and either make an additional redaction covering it ... or increase the image bbox accordingly if you find a pattern for how these captions are positioned relative to the image.

**Comment by megh-khaire** [comment]
By labels here I mean text printed on the image.
Example:
![example](https://user-images.githubusercontent.com/64032288/224492879-1fa5897d-faa5-4a6a-a814-067da6c105e9.png)

Thanks, @JorjMcKie I'll use redaction annot.
What garbage collection option would you recommend here?

**Reply by JorjMcKie** [reply]
>  What garbage collection option would you recommend here?

`.ez_save()` should do it. It uses garbage=3, deflate=True.

Your "labels" may be regular text outside the image - which would make it probably hard to find a programmatic way to include it.

**Reply by JorjMcKie** [reply]
BTW did you already confirm that these things in fact are images? Could aos be vector graphics ("line art"), which cannot be removed easily.

**Reply by megh-khaire** [reply]
@JorjMcKie yes they are images and the text is overlayed on these images. 
Redaction annot worked for me.

Thanks a lot for your prompt replies and amazing work 🙇
